packages/twenty-front/src/modules/favorites/hooks/useReorderFavorite.ts

```diff
@@ -11,28 +13,106 @@ export const useReorderFavorite = () => {
   const { updateOneRecord: updateOneFavorite } = useUpdateOneRecord({
     objectNameSingular: CoreObjectNameSingular.Favorite,
   });
+  const setActiveFavoriteFolderId = useSetRecoilState(
+    activeFavoriteFolderIdState,
+  );
 
   const handleReorderFavorite: OnDragEndResponder = (result) => {
-    if (!result.destination) return;
+    const { destination, source, draggableId } = result;
 
-    const draggedFavoriteId = result.draggableId;
-    const draggedFavorite = favorites.find((f) => f.id === draggedFavoriteId);
+    if (!destination) return;
 
+    if (
+      destination.droppableId === source.droppableId &&
+      destination.index === source.index
+    ) {
+      return;
+    }
+
+    const draggedFavorite = favorites.find((f) => f.id === draggableId);
     if (!draggedFavorite) return;
 
-    const inSameFolderFavorites = favoritesSorted.filter(
-      (fav) => fav.favoriteFolderId === draggedFavorite.favoriteFolderId,
+    if (destination.droppableId.startsWith('folder-header-')) {
+      const targetFolderId = destination.droppableId.replace(
+        'folder-header-',
+        '',
+      );
+      const folderFavorites = favoritesSorted.filter(
+        (favorite) => favorite.favoriteFolderId === targetFolderId,
+      );
+
+      let newPosition;
+      if (folderFavorites.length === 0) {
+        newPosition = 0;
+      } else {
+        newPosition = folderFavorites[folderFavorites.length - 1].position + 1;
+      }
+
+      updateOneFavorite({
+        idToUpdate: draggableId,
+        updateOneRecordInput: {
+          favoriteFolderId: targetFolderId,
+          position: newPosition,
+        },
+      });
+
+      setActiveFavoriteFolderId(targetFolderId);
+      return;
+    }
+
+    if (destination.droppableId !== source.droppableId) {
+      const newFolderId =
+        destination.droppableId === 'orphan-favorites'
+          ? null
+          : destination.droppableId.replace('folder-', '');
+
+      const destinationFavorites = favoritesSorted.filter(
+        (favorite) => favorite.favoriteFolderId === newFolderId,
+      );
+
+      let newPosition;
+      if (destinationFavorites.length === 0) {
```

Comment:
this position things is already present in getDraggedRecordPosition, can we re-use it?